File: apps/fip-web/src/app/features/flights/models/flight-summary.ts
export interface FlightSummary {
  flightId: string;
  callsign: string | null;
  icao24: string;
  startTime: string;
  endTime: string;
  duration: string;
  maximumAltitudeFeet: number | null;
  maximumGroundSpeedKnots: number | null;
  averageGroundSpeedKnots: number | null;
  maximumVerticalRateFeetPerMinute: number | null;
  minimumVerticalRateFeetPerMinute: number | null;
  distanceTraveledNauticalMiles: number;
  takeoffTime: string | null;
  landingTime: string | null;
  flightTime: string | null;
}

File: src/Core/Fip.Domain/Flights/FlightSummary.cs
namespace Fip.Domain.Flights;

/// <summary>
/// Statistics calculated from a reconstructed flight and its detected events.
/// </summary>
public sealed class FlightSummary
{
    public TimeSpan Duration { get; init; }

    public double DistanceNauticalMiles { get; init; }

    public double? MaximumAltitudeFeet { get; init; }

    public double? MaximumGroundSpeedKnots { get; init; }

    public double? AverageGroundSpeedKnots { get; init; }

    public double? MaximumClimbRate { get; init; }

    public double? MaximumDescentRate { get; init; }

    public DateTimeOffset? TakeoffTime { get; init; }

    public DateTimeOffset? LandingTime { get; init; }

    public TimeSpan? FlightTime { get; init; }
}
